apollo-router/src/plugins/demand_control/basic_cost_calculator.rs

```diff
@@ -127,9 +127,17 @@ impl BasicCostCalculator {
         schema: &Valid<Schema>,
     ) -> Result<f64, DemandControlError> {
         let mut cost = if operation.is_mutation() { 10.0 } else { 0.0 };
+
+        let Some(root_type_name) = schema.root_operation(operation.operation_type) else {
+            return Err(DemandControlError::QueryParseFailure(format!(
+                "Cannot cost {} operation because the schema does not support this root type",
+                operation.operation_type
+            )));
+        };
+
         cost += BasicCostCalculator::score_selection_set(
             &operation.selection_set,
-            operation.name.as_ref(),
+            root_type_name,
```

Comment:
this was just a small mistake, the operation name isn't a type name, the type is always the root type (eg. for any `query` it's always `Query`, or whatever the schema has specified)